ThunderCats/UIColor+TCAdditions.m

```diff
@@ -28,27 +28,42 @@
 
 
 #import "UIColor+TCAdditions.h"
+#import "TCInvalidArgument.h"
 
 @implementation UIColor (TCAdditions)
 
 + (UIColor *)tc_colorWithHexString:(NSString *)hexString
 {
-    NSString *cleanString = [hexString stringByReplacingOccurrencesOfString:@"#" withString:@""];
-    if ([cleanString length] == 3)
+    if([hexString hasPrefix:@"#"])
     {
-        cleanString = [NSString stringWithFormat:@"%@%@%@%@%@%@",
-                       [cleanString substringWithRange:NSMakeRange(0, 1)], [cleanString substringWithRange:NSMakeRange(0, 1)],
-                       [cleanString substringWithRange:NSMakeRange(1, 1)], [cleanString substringWithRange:NSMakeRange(1, 1)],
-                       [cleanString substringWithRange:NSMakeRange(2, 1)], [cleanString substringWithRange:NSMakeRange(2, 1)]];
+        hexString = [hexString substringFromIndex:1];
     }
     
-    if ([cleanString length] == 6)
+    NSCharacterSet *validCharacters = [NSCharacterSet characterSetWithCharactersInString:@"0123456789ABCDEFabcdefg"];
+    
+    bool invalidLength = [hexString length] != 3 && [hexString length] != 6;
```

Comment:
4 & 8 should also count as valid lengths.  Hex strings with a length of 4 or 8 are also specifying the alpha value of the color.